packages/samples/headless-react/src/components/category-facet/category-facet.class.tsx

```diff
@@ -1,34 +1,35 @@
-import {Component} from 'react';
+import {Component, ContextType} from 'react';
 import {
   buildCategoryFacet,
   CategoryFacet as HeadlessCategoryFacet,
   CategoryFacetState,
   CategoryFacetValue,
   Unsubscribe,
 } from '@coveo/headless';
-import {engine} from '../../engine';
 import {CategoryFacetSearch} from './category-facet-search';
+import {AppContext} from '../../context/engine';
 
 interface CategoryFacetProps {
   field: string;
   facetId: string;
 }
 
-export class CategoryFacet extends Component<CategoryFacetProps> {
-  private controller: HeadlessCategoryFacet;
-  public state: CategoryFacetState;
-  private unsubscribe: Unsubscribe = () => {};
+export class CategoryFacet extends Component<
+  CategoryFacetProps,
+  CategoryFacetState
```

Comment:
The type for the state here isn't entirely accurate since `this.state` is initially `undefined`. I could change it to `CategoryFacetState | undefined` if you prefer.

Reply:
It would improve it a bit, since otherwise it's implied that even on the first call of `render`, `this.state` exists (which isn't the case).

Reply:
update: I checked, and improving type safety like this would involve another relatively big change and doesn't really seem to be worth it.